hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -424,9 +159,7 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() throws IOException {
                   LESSER_THAN, oldestPendingInstant.get().getTimestamp())).findFirst());
       // Check if the completed instant is higher than the oldest inflight instant
       // in that case update the oldestCommitToRetain to oldestInflight commit time.
-      if (!completedCommitBeforeOldestPendingInstant.isPresent()
-          || HoodieTimeline.compareTimestamps(oldestPendingInstant.get().getTimestamp(),
-          LESSER_THAN, completedCommitBeforeOldestPendingInstant.get().getTimestamp())) {
+      if (!completedCommitBeforeOldestPendingInstant.isPresent()) {
```

Comment:
why do we need this change? this may break sth. please revert if unnecessary for this change.

Reply:
Because the check
```java
HoodieTimeline.compareTimestamps(oldestPendingInstant.get().getTimestamp(),
          LESSER_THAN, completedCommitBeforeOldestPendingInstant.get().getTimestamp())
```

is always false. I can revert the change.